data/state/accountsDB.go

```diff
@@ -584,15 +636,42 @@ func (adb *AccountsDB) CancelPrune(rootHash []byte, identifier data.TriePruningI
 // SnapshotState triggers the snapshotting process of the state trie
 func (adb *AccountsDB) SnapshotState(rootHash []byte) {
 	log.Trace("accountsDB.SnapshotState", "root hash", rootHash)
-
+	adb.mainTrie.EnterSnapshotMode()
 	adb.mainTrie.TakeSnapshot(rootHash)
+
+	go func() {
+		adb.snapshotUserAccountDataTrie(rootHash)
+		adb.mainTrie.ExitSnapshotMode()
+	}()
+}
+
+func (adb *AccountsDB) snapshotUserAccountDataTrie(rootHash []byte) {
+	leafs, err := adb.GetAllLeaves(rootHash)
+	if err != nil {
+		log.Error("incomplete snapshot as getAllLeaves error", "error", err)
+		return
+	}
+
+	for _, leaf := range leafs {
+		account := &Account{}
+		err = adb.marshalizer.Unmarshal(account, leaf)
+		if err != nil {
+			log.Trace("this must be a leaf with code", "err", err)
```

Comment:
Code is saved as a []byte in the leaf. When accessing the leaf there is no information whether it is a code / account. So first there is a try to unmarshal into an account, otherwise it must be code.